MINOR: Safe update dependencies

Mirai1129 opened the pull request:
These dependencies have been updated across both files:

    caffeine: From 3.1.8 to 3.2.0      javassist: From 3.29.2-GA to
3.30.2-GA      Jetty-related: All Jetty components have been updated
from 12.0.15 to 12.0.22, including:          jetty-alpn-client
jetty-client          jetty-ee10-servlet          jetty-ee10-servlets
jetty-http          jetty-io          jetty-security
jetty-server          jetty-session          jetty-util      jose4j:
From 0.9.4 to 0.9.6      Jersey-related: All Jersey components have been
updated from 3.1.9 to 3.1.10, including:          jersey-client
jersey-common          jersey-container-servlet
jersey-container-servlet-core          jersey-hk2          jersey-server
classgraph: From 4.8.173 to 4.8.179      jline: From 3.25.1 to 3.30.4
pcollections: From 4.0.1 to 4.0.2      re2j: From 1.7 to 1.8
snappy-java: From 1.1.10.5 to 1.1.10.7

New Dependency (LICENSE-binary only)

    A new dependency, jspecify-1.0.0, has been added to LICENSE-binary.

gradle/dependencies.gradle Specific Updates

These updates are only reflected in the gradle/dependencies.gradle file:

    bcpkix: From 1.78.1 to 1.80      bndlib: From 7.0.0 to 7.1.0 jacoco:
From 0.8.10 to 0.8.13      hamcrest: From 2.2 to 3.0      jqwik: From
1.8.3 to 1.9.2

Reviewers: Ken Huang <s7133700@gmail.com>, Chia-Ping Tsai
 <chia7712@gmail.com>

chia7712 commented:
@Mirai1129 please update the license files too

m1a2st commented on gradle/dependencies.gradle:
```diff
@@ -109,23 +109,23 @@ versions += [
-  rocksDB: "9.7.3",
+  pcollections: "4.0.2",
+  re2j: "1.8",
+  rocksDB: "9.7.4",
```
I think we don’t need to update the RocksDB minor version in this PR, since another PR will handle the major version upgrade.

Ref: https://github.com/apache/kafka/pull/19880/files

m1a2st commented on gradle/dependencies.gradle:
```diff
@@ -53,33 +53,33 @@ versions += [
   jopt: "5.0.4",
-  jose4j: "0.9.4",
+  jose4j: "0.9.6",
   junit: "5.10.2",
```
```suggestion
  junit: "5.13.0",
```

m1a2st commented on gradle/dependencies.gradle:
```diff
@@ -109,23 +109,23 @@ versions += [
   mavenArtifact: "3.9.6",
   metrics: "2.2.0",
   mockito: "5.14.2",
```
```suggestion
  mockito: "5.18.0",
```

m1a2st reviewed (commented):
Thanks @Mirai1129 for this patch, Could you check if the upgraded version is the latest available minor version under the current major version?

Mirai1129 commented:
Thanks for the suggestion @m1a2st . I've updated the dependencies with safe version bumps in the latest commit.

Mirai1129 committed:
safe update dependencies

Mirai1129 committed:
downgrade junit version

chia7712 commented:
@Mirai1129 Please revert the update from junit-releated libraries. I have opened https://issues.apache.org/jira/browse/KAFKA-19382 to trace it

Mirai1129 committed:
upgrade junit version

Mirai1129 committed:
revert the update from junit-releated libraries.

chia7712 commented on gradle/dependencies.gradle:
```diff
@@ -108,25 +108,25 @@ versions += [
-  zstd: "1.5.6-10",
-  junitPlatform: "1.10.2",
+  zstd: "1.5.7-3",
+  junitPlatform: "1.13.0",
```
please revert this change

Mirai1129 committed:
revert the update from junit-releated libraries.

Mirai1129 committed:
revert zstd libraries.

Mirai1129 committed:
revert zstd libraries.

Mirai1129 committed:
Merge branch 'trunk' into MINOR-safe-update-dependencies

YutaLin committed:
KAFKA-18486 Update activeProducerState wih KRaft mechanism in ReplicaManagerTest (#19890)

Description:
* replace RPC with KRaft mechanism to test activeProducerState in
ReplicaManagerTest

Reviewers: TaiJuWu <tjwu1217@gmail.com>, Chia-Ping Tsai
 <chia7712@gmail.com>

jingjia88 committed:
KAFKA-18486 Update testExceptionWhenUnverifiedTransactionHasMultipleProducerIds  (#19883)

- Replace the deprecated `becomeLeaderOrFollower` with the
metadata-based `applyDelta` method.
- Add overloaded `topicsCreateDelta` to support custom topic name and
topicId.

Reviewers: Ken Huang <s7133700@gmail.com>, TengYao Chi
 <kitingiao@gmail.com>, Nick Guo <lansg0504@gmail.com>, Chia-Ping Tsai
 <chia7712@gmail.com>

m1a2st committed:
KAFKA-19042 Move PlaintextConsumerSubscriptionTest to client-integration-tests module (#19827)

Use Java to rewrite PlaintextConsumerSubscriptionTest by new test infra
and move it to client-integration-tests module.

Reviewers: Chia-Ping Tsai <chia7712@gmail.com>

chia7712 commented on gradle/dependencies.gradle:
```diff
@@ -108,19 +108,19 @@ versions += [
-  mockito: "5.14.2",
-  opentelemetryProto: "1.0.0-alpha",
+  mockito: "5.18.0",
+  opentelemetryProto: "1.7.0-alpha",
```
this is not a small update. Could you please revert it to keep the update "safe"?

chia7712 replied:
open https://issues.apache.org/jira/browse/KAFKA-19391 for it

Mirai1129 committed:
revert mockito and opentelemetryProto

Mirai1129 committed:
Update LICENSE-binary

Mirai1129 committed:
Merge branch 'trunk' into MINOR-safe-update-dependencies

m1a2st reviewed (approved):
Thanks, LGTM
